testsuite/cli/src/client_proxy.rs

```diff
@@ -598,13 +599,17 @@ impl ClientProxy {
         )?;
         self.client.submit_transaction(Some(&mut sender), txn)?;
         if is_blocking {
-            self.wait_for_transaction(sender.address, sender.sequence_number);
+            self.wait_for_transaction(sender.address, sender.sequence_number)?;
         }
         Ok(())
     }
 
     /// Waits for the next transaction for a specific address and prints it
-    pub fn wait_for_transaction(&mut self, account: AccountAddress, sequence_number: u64) {
+    pub fn wait_for_transaction(
+        &mut self,
+        account: AccountAddress,
+        sequence_number: u64,
+    ) -> Result<()> {
         let mut max_iterations = 5000;
```

Comment:
Instead of using fixed iteration, could we set an time deadline instead?  because the defualt timeout settings on the JsonRpcCient. this could be super long if we wait for 5000 * (max_timout + 10ms)

Reply:
All I'm changing here is to return a Result from wait_for_transaction. I see your point about using a time deadline, but that should be a separate change (preferably done by someone more familiar with the history of this code).